tests/functional/s3/test_cp_command.py

```diff
@@ -760,6 +760,72 @@ def test_cp_with_error_and_warning_permissions(self):
         self.assertIn('upload failed', stderr)
         self.assertIn('warning: File has an invalid timestamp.', stderr)
 
+    def test_upload_with_flexible_checksum_crc32(self):
```

Comment:
Let's avoid using the term "flexible checksum". Instead, refer to the parameter names `checksum_algorithm` and `checksum_mode`. This is mainly to clarify what exactly the test is for.